domain/machine/state/reboot.go

```diff
@@ -201,3 +193,22 @@ func (st *State) ShouldRebootOrShutdown(ctx context.Context, uuid string) (machi
 	}
 	return machine.ShouldDoNothing, nil
 }
+
+// InitialWatchMachineRebootStatement returns the table and the initial watch statement
+// for watching machine reboot changes.
+//
+// It collects all uuid from `machine_requires_reboot` table and join all uuid from possible child of those
+// machine from `machine_parent` table
+//
+// Disclaimer: This statement assumes that grandparent are not a thing, by design
+// If the design changes, we need to update this statement.
+// Here, we imply that if a reboot is required for a machine, an event is also generated for its
+// child.
+func (st *State) InitialWatchMachineRebootStatement() (string, string) {
+	return "machine_requires_reboot", `
+		SELECT DISTINCT machine_uuid AS uuid
+		FROM machine_requires_reboot
+		UNION SELECT DISTINCT parent.machine_uuid as uuid
+		FROM machine_parent parent JOIN machine_requires_reboot reboot on reboot.machine_uuid = parent.parent_uuid
```

Comment:
This should be left aligned. It's what we're doing currently in the state package (i.e. column 0).